src/R/Support/Impl/Help/Packages/PackageIndex.cs

```diff
@@ -69,15 +70,16 @@ public sealed class PackageIndex : IPackageIndex {
             }
         }
 
-        private Task OnBeforePackagesInstalledAsync(object sender, EventArgs e) {
+        public Task BeforePackagesInstalledAsync(CancellationToken ct) {
             try {
-                _host.StopSessionAsync().Wait(3000);
+                _host.StopSessionAsync(ct).Wait(3000);
```

Comment:
No, call should be async. If you need a timeout, you can make a `cancellationToken` with timeout:
```cs
var timeoutCts = CancellationTokenSource.CreateLinkedTokenSource(ct);
timeoutCts.CancelAfter(3000);
await _host.StopSessionAsync(timeoutCts.Token);
```